sqlmesh/schedulers/airflow/plugin.py

```diff
@@ -28,14 +31,16 @@ def on_load(cls, *args: t.Any, **kwargs: t.Any) -> None:
             logger.info("MWAA Webserver instance detected. Skipping SQLMesh state migration...")
             return
 
-        default_catalog = Variable.get(DEFAULT_CATALOG_VARIABLE_NAME, default_var=None)
+        default_catalog = Variable.get(DEFAULT_CATALOG_VARIABLE_NAME, default_var="MISSING")
+        if default_catalog == NO_DEFAULT_CATALOG_VALUE:
+            default_catalog = None
 
         with util.scoped_state_sync() as state_sync:
             try:
                 # If default catalog is not defined then we want to raise an error unless
                 # this is a fresh install since we know nothing needs to be migrated and
                 # the client will prevent making any changes until the default catalog is set.
-                if not default_catalog:
+                if default_catalog == "MISSING":
```

Comment:
Do we need `MISSING`? Should we just do 
```
if default_catalog == NO_DEFAULT_CATALOG_VALUE:
            default_catalog = None
```
right after this check?

Reply:
I believe the issue would be in the try/except. If we did that then you could end up at `versions = state_sync.get_versions(validate=False)` and throw an exception which would then end up migrating with a default catalog value that is not correct. Therefore you would need to duplicate that logic to convert the no default catalog to none.

Reply:
ok, let's come up with a more distinct placeholder then and maybe leave a comment. This is a good gotcha and it wasn't obvious to me at all. I can totally see someone using `MISSING` as a catalog name lol.

Reply:
Let me know what you think: https://github.com/TobikoData/sqlmesh/pull/2881/commits/2e578b965e88c3ef94210e8dbf72d84d4f2067d8